mlx-rs/Cargo.toml

```diff
@@ -35,6 +35,8 @@ libc = "0.2"
 [dev-dependencies]
 pretty_assertions.workspace = true
 float_eq.workspace = true
+mlx-nn.workspace = true
```

Comment:
I also thought this would incur circular dependency, but it turns out to be fine. I think this is because the dev-deps are used for building the tests, which are built as independent binaries, and both mlx-rs and mlx-nn would appear as deps to these tests

Reply:
cargo would actually complain if `mlx-nn` is moved from the dev-dependencies section to the dependencies section